GVFS/GVFS.Service/Mac/GVFSService.cs

```diff
@@ -0,0 +1,115 @@
+using GVFS.Common;
+using GVFS.Common.NamedPipes;
+using GVFS.Common.Tracing;
+using System;
+using System.Linq;
+using System.Threading;
+
+namespace GVFS.Service
+{
+    public class GVFSService
+    {
+        private const string ServiceNameArgPrefix = "--servicename=";
+        private const string EtwArea = nameof(GVFSService);
+
+        private JsonTracer tracer;
+        private Thread serviceThread;
+        private ManualResetEvent serviceStopped;
+        private string serviceName;
+        private string serviceDataLocation;
+
+        public GVFSService(JsonTracer tracer)
+        {
+            this.tracer = tracer;
+            this.serviceName = GVFSConstants.Service.ServiceName;
+        }
+
+        public void RunWithArgs(string[] args)
+        {
+            if (this.serviceThread != null)
+            {
+                throw new InvalidOperationException("Cannot start service twice in a row.");
+            }
+
+            string nameArg = args.FirstOrDefault(arg => arg.StartsWith(ServiceNameArgPrefix, StringComparison.OrdinalIgnoreCase));
+            if (nameArg != null)
+            {
+                this.serviceName = nameArg.Substring(ServiceNameArgPrefix.Length);
+            }
+
+            try
+            {
+                this.serviceDataLocation = Paths.GetServiceDataRoot(this.serviceName);
+                this.StartServiceThread();
+                this.serviceThread.Join();
+            }
+            catch (Exception e)
+            {
+                this.LogExceptionAndExit(e, nameof(this.RunWithArgs));
+            }
+        }
+
+        private void StartServiceThread()
+        {
+            if (this.serviceStopped != null)
+            {
+                return;
+            }
+
+            this.serviceStopped = new ManualResetEvent(false);
+            this.serviceThread = new Thread(this.ServiceThreadMain);
+
+            this.serviceThread.Start();
+        }
+
+        private void ServiceThreadMain()
+        {
+            try
+            {
+                EventMetadata metadata = new EventMetadata();
+                metadata.Add("Version", ProcessHelper.GetCurrentProcessVersion());
+                this.tracer.RelatedEvent(EventLevel.Informational, $"{nameof(GVFSService)}_{nameof(this.ServiceThreadMain)}", metadata);
+
+                string pipeName = this.serviceName + ".Pipe";
+                this.tracer.RelatedInfo("Starting pipe server with name: " + pipeName);
+
+                using (NamedPipeServer pipeServer = NamedPipeServer.StartNewServer(pipeName, this.tracer, this.HandleRequest))

```

Comment:
Do we need/want to start the named pipe server in this iteration of GVFS.Service on Mac?

Also, once we know where GVFS.Service will store it's config files we should probably switch to using [a rooted pipe name](https://github.com/Microsoft/VFSForGit/blob/d02c62832d512ab3ef91aeed5d0457b4675934da/GVFS/GVFS.Platform.Mac/MacPlatform.Shared.cs#L35) to make the pipe more easily discoverable.

Reply:
We don't need it now. Can add it later with rooted pipe name.